libred/red.c

```diff
@@ -459,6 +477,132 @@ static int call_posix_spawnp(pid_t *restrict pid, const char *restrict file,
 }
 #endif
 
+
+#define redirect_hardcoded(path, replacement)                         \
+  patterns[0] = "^/usr/bin/" path "$";                                \
+  patterns[1] = "^/bin/" path "$";                                    \
+  patterns[2] = "^/usr/sbin/" path "$";                               \
+  for(int i = 0; i < 3; ++i){                                         \
+    compiled = pcre_compile(patterns[i], 0, &err_ptr,                 \
```

Comment:
You're compiling the pattern on every invocation while this could be done just once during initialization. However, like I mentioned in the previous comment, using PCRE here is really an overkill.